### PR TITLE
chore(dep)!: upgrade typescript

### PR DESCRIPTION
**Pull request by summer-ji-eng**
upgrade 

- typescript version `^3.8.3`

- gts version `2.0.0`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,14 +36,14 @@
     "@types/mocha": "^7.0.0",
     "@types/node": "^10.5.2",
     "@types/sinon": "^7.0.5",
+    "c8": "^7.0.0",
     "codecov": "^3.0.4",
-    "gts": "^1.0.0",
+    "gts": "2.0.0-alpha.5",
```
Comment (JustinBeckwith) [inline package.json]: Could just pin this to `next` for now to always float to the latest

File: package.json [hunk 2]
```diff
@@ -36,14 +36,14 @@
...
     "mocha": "^7.0.0",
-    "c8": "^7.0.0",
     "sinon": "^9.0.0",
-    "typescript": "3.6.4"
+    "typescript": "^3.8.3"
   },
   "engines": {
-    "node": ">=10"
+    "node": ">=10.4.0"
```
Comment (JustinBeckwith) [inline package.json]: Is this the story across the board?  `10.4.0`?

Comment (summer-ji-eng) [reply package.json]: BigInt is suitable until 10.4.0

File: package.json (hunk 1 above)
Comment (bcoe) [reply package.json]: I think we're getting very close to being able to just cut `2.x.x`, I'd rather we're explicit about version, and let renovate do its thing soon.

**Pull request merged**